src/RandomEnchants.cpp

```diff
@@ -1,155 +1,214 @@
-		switch (Class)
-		{
-		case 2:
-			ClassQueryString = "WEAPON";
-			break;
-		case 4:
-			ClassQueryString = "ARMOR";
-			break;
-		}
-		if (ClassQueryString == "")
-			return -1;
-		uint32 Quality = item->GetTemplate()->Quality;
-		int rarityRoll = -1;
-		switch (Quality)
-		{
-		case 0://grey
-			rarityRoll = rand_norm() * 25;
-			break;
-		case 1://white
-			rarityRoll = rand_norm() * 50;
-			break;
-		case 2://green
-			rarityRoll = 45 + (rand_norm() * 20);
-			break;
-		case 3://blue
-			rarityRoll = 65 + (rand_norm() * 15);
-			break;
-		case 4://purple
-			rarityRoll = 80 + (rand_norm() * 14);
-			break;
-		case 5://orange
-			rarityRoll = 93;
-			break;
-		}
-		if (rarityRoll < 0)
-			return -1;
-		int tier = 0;
-		if (rarityRoll <= 44)
-			tier = 1;
-		else if (rarityRoll <= 64)
-			tier = 2;
-		else if (rarityRoll <= 79)
-			tier = 3;
-		else if (rarityRoll <= 92)
-			tier = 4;
-		else
-			tier = 5;
-		
-		QueryResult qr = WorldDatabase.PQuery("SELECT enchantID FROM item_enchantment_random_tiers WHERE tier='%d' AND exclusiveSubClass=NULL AND class='%s' OR exclusiveSubClass='%u' OR class='ANY' ORDER BY RAND() LIMIT 1", tier, ClassQueryString.c_str(), item->GetTemplate()->SubClass);
-		return qr->Fetch()[0].GetUInt32();
-	}
-};
-class RandomEnchantsWorld : public WorldScript
+class REConfig : public WorldScript
 {
 public:
-    RandomEnchantsWorld() : WorldScript("RandomEnchantsWorld") { }
+    REConfig() : WorldScript("REConfig") { }
 
     void OnBeforeConfigLoad(bool reload) override
     {
         if (!reload) {
             std::string conf_path = _CONF_DIR;
-            std::string cfg_file = conf_path + "/RandomEnchants.conf";
+            std::string cfg_file = conf_path + "/mod_randomenchants.conf";
 
-			std::string cfg_def_file = cfg_file +".dist";
+            std::string cfg_def_file = cfg_file + ".dist";
             sConfigMgr->LoadMore(cfg_def_file.c_str());
-
             sConfigMgr->LoadMore(cfg_file.c_str());
+
+            // Load Configuration Settings
+            SetInitialWorldSettings();
         }
     }
+
+    // Load Configuration Settings
+    void SetInitialWorldSettings()
+    {
+        RandomEnchantEnabled = sConfigMgr->GetBoolDefault("RandomEnchants.Enabled", true);
+        RandomEnchantAnnounce = sConfigMgr->GetBoolDefault("RandomEnchants.Announce", true);
+        Crafted = sConfigMgr->GetBoolDefault("RandomEnchants.OnCreate", 1);
+        Looted = sConfigMgr->GetBoolDefault("RandomEnchants.OnLoot", 1);
+        QuestReward = sConfigMgr->GetBoolDefault("RandomEnchants.OnQuestReward", 1);
+        HighQuality = sConfigMgr->GetIntDefault("RandomEnchants.HighQuality", 1);
+        LowQuality = sConfigMgr->GetIntDefault("RandomEnchants.LowQuality", 1);
+        Chance1 = sConfigMgr->GetIntDefault("RandomEnchants.Chance1", 60);
+        Chance2 = sConfigMgr->GetIntDefault("RandomEnchants.Chance2", 65);
+        Chance3 = sConfigMgr->GetIntDefault("RandomEnchants.Chance3", 70);
```

Comment:
ok so hmm, the defautl values here, they will only happen if they are not found in randomenchants.conf.dist which is very unlikely

so why let them here?

I feel like it's better to remove them from the code and have them only defined in the conf

What do you think @milestorme @wizzymore ? (i know it's a detail)

Reply:
milestorme, you resolved the conv but didn't answer lol

Reply:
if You want , doesn’t really matter .

Reply:
@wizzymore see